src/Components/shared/Form.tsx

```diff
@@ -100,22 +103,75 @@ export const UserForm: React.FC<UserFormProps> = ({ open, onClose, initialValues
               flexDirection: 'column',
               gap: '1rem'
             }}>
-              <Field as={TextField} name="fullName" label="Full Name" required />
-              <Field as={TextField} name="email" label="Email" required />
-              {mode === 'add' && <Field as={TextField} name="password" label="Password" type="password" required />}
-              <Box sx={{ display: 'flex', justifyContent: 'center', margin: '1rem 0', gap: '2rem' }}>
-                <Button type="submit" onClick={() => formik.handleSubmit()}>
+              <Field
+                as={TextField}
+                name="fullName"
+                label="Full Name"
+                required
+                error={formik.touched.fullName && Boolean(formik.errors.fullName)}
+                helperText={formik.touched.fullName && formik.errors.fullName}
+              />
+
+              <Field
+                as={TextField}
+                name="email"
+                label="Email"
+                required
+                error={formik.touched.email && Boolean(formik.errors.email)}
+                helperText={formik.touched.email && formik.errors.email}
+              />
+
+              {mode === 'add' && 'password' in formik.values && (
+                <>
```

Comment:
I think that `mode === 'add'` should suffice. it's redundant to also check the password in `values`